### PR TITLE
Updates to RS_Ellipse::scale

### PR DESCRIPTION
**Pull request by Ngassa**
I've done more fixes to rs_ellipse.cpp file to make sure LibreCAD builds.

### CODE REVIEW

File: librecad/src/lib/engine/rs_ellipse.cpp [hunk 1]
```diff
@@ -1324,7 +1324,7 @@ const RS_EllipseData& RS_Ellipse::getData() const
  * x *= kx
  * y *= ky
  * find the maximum and minimum of x^2 + y^2,
- */
```
Comment (dxli) [inline librecad/src/lib/engine/rs_ellipse.cpp]: this won't compile

File: librecad/src/lib/engine/rs_ellipse.cpp [hunk 2]
```diff
@@ -1372,6 +1372,36 @@ void RS_Ellipse::scale(const RS_Vector& center, const RS_Vector& factor) {
     scaleBorders(center,factor);
 // calculateBorders();
 
+}*/
+
+
+void RS_Ellipse::scale(const RS_Vector& center, const RS_Vector& factor)
+{
+    data.center.scale(center, factor);
```
Comment (dxli) [inline librecad/src/lib/engine/rs_ellipse.cpp]: this line should be removed. all scaling is by quadratic

File: librecad/src/lib/engine/rs_ellipse.cpp [hunk 3]
```diff
@@ -1372,6 +1372,36 @@ void RS_Ellipse::scale(const RS_Vector& center, const RS_Vector& factor) {
...
+
+    boost::numeric::ublas::matrix<double> eqn_Quad = eqn.getQuad();
+    double const& eqn_a = eqn_Quad(0,0);
+    double const& eqn_c=2.* eqn_Quad(0,1);
+    double const& eqn_b = eqn_Quad(1,1);
+    
+    // quadratic equation of scaled ellipse
+    LC_Quadratic scaled_eqn = eqn;
+ 
+    scaled_eqn.m_mQuad(0,0) = eqn_a * factor.y * factor.y;
```
Comment (dxli) [inline librecad/src/lib/engine/rs_ellipse.cpp]: the scaling is correct. Please carry out test to show the old and new scale() methods generate the same results for several test cases. Please design several test cases for unit testing.

File: librecad/src/lib/engine/rs_ellipse.cpp (hunk 1 above)
Comment (Ngassa) [inline librecad/src/lib/engine/rs_ellipse.cpp]: I just wanted to comment away the former scale() member function. This LibreCAD code compiled on my machine before I went on to make a pull request.

File: librecad/src/lib/engine/rs_ellipse.cpp (hunk 2 above)
Comment (Ngassa) [inline librecad/src/lib/engine/rs_ellipse.cpp]: Thanks very much for the reminder Donhxu Li.

File: librecad/src/lib/engine/rs_ellipse.cpp (hunk 3 above)
Comment (Ngassa) [inline librecad/src/lib/engine/rs_ellipse.cpp]: Thanks very much for the review base.

**Comment by dxli** [comment]
Please keep your code in a separate git branch for the time being.

File: librecad/src/lib/engine/rs_ellipse.cpp (hunk 1 above)
Comment (dxli) [inline librecad/src/lib/engine/rs_ellipse.cpp]: To do carry out testing, please keep both methods(renaming them?)

Try something like, (just ideas, not correct at coding)

cout << ellipse.oldscale().getangle()-ellipse.scale().getangle()


cout << ellipse.oldscale().geratio()-ellipse.scale().geratio()

**Pull request closed without merging**